local_example_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/gonum/optimize"
+	"github.com/gonum/optimize/functions"
+)
+
+func ExampleLocal_BFGS() {
+	p := optimize.Problem{
+		Func: functions.ExtendedRosenbrock{}.Func,
+		Grad: functions.ExtendedRosenbrock{}.Grad,
+	}
+
+	method := &optimize.BFGS{}
+
+	x := []float64{1.3, 0.7, 0.8, 1.9, 1.2}
+	settings := optimize.DefaultSettings()
+	settings.Recorder = nil
+	settings.GradientThreshold = 1e-12
+	settings.FunctionConverge = nil
+
+	result, err := optimize.Local(p, x, settings, method)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err = result.Status.Err(); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("result.Status: %v\n", result.Status)
+	fmt.Printf("result.X: %v\n", result.X)
+	fmt.Printf("result.F: %v\n", result.F)
+	// Output:
```

Comment:
Maybe add the number of function evaluations? That's going to be the relevant quantity between methods.